x/evm/handler.go

```diff
@@ -113,15 +113,18 @@ func NewHandler(k types.BaseKeeper, t types.TSS, v types.Voter, s types.Signer,
 			res, err := server.SignCommands(sdk.WrapSDKContext(ctx), msg)
 			result, err := sdk.WrapServiceResult(ctx, res, err)
 			if err == nil {
-				result.Log = fmt.Sprintf("successfully started signing batched commands with ID %s", hex.EncodeToString(res.BatchedCommandsID))
+				if res.BatchedCommandsID != nil {
+					result.Log = fmt.Sprintf("successfully started signing batched commands with ID %s", hex.EncodeToString(res.BatchedCommandsID))
+				} else {
+					result.Log = "no command to sign found"
```

Comment:
is it fine to `return result, err` if `res.BatchedCommandsID != nil`?

Reply:
I suggest you change here https://github.com/axelarnetwork/axelar-core/blob/ea46c0233b4d79d146cd74d4b65fe78d8e481ec2/proto/evm/v1beta1/tx.proto#L211 to add something like command_count into the response. `nil` as `bytes` in protobuf would be serialized to empty bytes which isn't very good as an API.

Reply:
good point, added a new field in `SignCommandsResponse`.